realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -2529,17 +2548,45 @@ public void isNull_linkField() {
         // 10 Date
         assertEquals(2, realm.where(NullTypes.class).isNull(
                 NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_DATE_NULL).count());
+    }
+
+    @Test
+    public void isNull_unsupportedLinkedTypes() {
+        RealmQuery<NullTypes> result;
+
         // FIXME
-        // Core now supports `isNotNull` on nested link fields: https://github.com/realm/realm-cocoa/pull/4743
+        // Core now supports `isNull` on nested link fields: https://github.com/realm/realm-cocoa/pull/4743
         // tracked in https://github.com/realm/realm-java/issues/4529
-        // 11 Object
-        // assertEquals(1, realm.where(NullTypes.class).isNull(
+        // 12 Object
+        //assertEquals(1, realm.where(NullTypes.class).isNull(
         //        NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_OBJECT_NULL).count());
         try {
-            realm.where(NullTypes.class).isNull(
-                    NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_OBJECT_NULL);
-            fail();
-        } catch (IllegalArgumentException ignored) {
+            result = realm.where(NullTypes.class).isNull(NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_OBJECT_NULL);
+            fail("isNull should throw on nested linked fields (OBJECT)");
+        } catch (IllegalArgumentException expected) {
+            assertEquals("Illegal Argument: isNull() by nested query for link field is not supported.", expected.getMessage());
+        }
+
+        // 13 LIST
+        try {
+            result = realm.where(NullTypes.class).isNull(NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_LIST_NULL);
+            fail("isNull should throw on nested linked fields (LIST)");
+        } catch (IllegalArgumentException expected) {
+            assertEquals("Illegal Argument: RealmList(fieldListNull) is not nullable.", expected.getMessage());
```

Comment:
Any reason we throw in this case, it is trivially `false` since nothing will ever match this criteria, but do we really want to throw here. I could see some use cases where generic code might end constructing this kind of queries? Same in the others below